src/lib.rs

```diff
@@ -69,75 +110,87 @@ pub unsafe extern "C" fn pthread_join(
     native: libc::pthread_t,
     _value: *mut *mut libc::c_void,
 ) -> libc::c_int {
-    if native == MAIN_THREAD_ID {
-        // This is not a valid thread to join on
-        return libc::EINVAL;
```

Comment:
This check should remain.

Reply:
Originally I was going to disagree, but after thinking about it more, I think maybe we should just be checking if `*(native as *mut PThread).thread.is_null()` rather than this check, although I suppose null ptr to `PThread` is also invalid, so maybe both checks are good.

Reply:
In the case where it's an invalid thread ID (not main), we should return ESRCH:

> ESRCH
    No thread with the ID thread could be found.

https://linux.die.net/man/3/pthread_join